### ISSUE TITLE
Errors executing an ios_static_framework imported by SwiftPM (missing Info.plist?)

### ISSUE THREAD

**Issue by jagobagascon** [bazelbuild/rules_apple#969]
Hi!

I'm looking for a way to compile and distribute our SDK for iOS using Bazel and Swift Package Manager.

The core of our library is built in C and our goal is to distribute it as a binary so that users do not have to compile it themselves. 
So far I managed to compile the core as an ios static framework for each of our supported architectures, then we bundle them in a xcframework:
```
cc_library(
    name = 'CORE',
    ....
)

ios_static_framework(
    name = 'ios_static_framework',
    deps = [ ':CORE' ],
    minimum_os_version = '10.0',
    hdrs = glob(['**/*.h']),
    bundle_name = module_name,
)
```
```
$  bazel build :ios_static_framework --ios_multi_cpus=arm64
$  bazel build :ios_static_framework --ios_multi_cpus=x86_64
$  xcodebuild -create-xcframework -framework ./arm64/Native.framework -framework ./x86_64/Native.framework -output Native.xcframework
```

This way I can import it as a Swift dependency on Xcode and compile our test application with no problems at all. However, **when I try to run that app I'm consistently getting the following error**:
```
This app could not be installed at this time.
Domain: IXUserPresentableErrorDomain
Code: 1
Failure Reason: Could not install at this time.
Recovery Suggestion: Failed to load Info.plist from bundle at path /Users/jagoba/Library/Developer/CoreSimulator/Devices/6756F0BC-6C41-410F-93B1-AAD2E69195F0/data/Library/Caches/com.apple.mobile.installd.staging/temp.GJj0xM/extracted/Payload/prueba.app/Frameworks/Native.framework; Extra info about Info.plist: Couldn't stat /Users/jagoba/Library/Developer/CoreSimulator/Devices/6756F0BC-6C41-410F-93B1-AAD2E69195F0/data/Library/Caches/com.apple.mobile.installd.staging/temp.GJj0xM/extracted/Payload/prueba.app/Frameworks/Native.framework/Info.plist: No such file or directory
--
Failed to load Info.plist from bundle at path /Users/jagoba/Library/Developer/CoreSimulator/Devices/6756F0BC-6C41-410F-93B1-AAD2E69195F0/data/Library/Caches/com.apple.mobile.installd.staging/temp.GJj0xM/extracted/Payload/prueba.app/Frameworks/Native.framework; Extra info about Info.plist: Couldn't stat /Users/jagoba/Library/Developer/CoreSimulator/Devices/6756F0BC-6C41-410F-93B1-AAD2E69195F0/data/Library/Caches/com.apple.mobile.installd.staging/temp.GJj0xM/extracted/Payload/prueba.app/Frameworks/Native.framework/Info.plist: No such file or directory
Domain: MIInstallerErrorDomain
Code: 35
User Info: {
    FunctionName = "-[MIBundle _validateWithError:]";
    LegacyErrorString = PackageInspectionFailed;
    SourceFileLine = 131;
}
--


System Information

macOS Version 10.15.6 (Build 19G2021)
Xcode 12.0.1 (17220)
```

I'm quite new to iOS development so I'm a bit lost on this one.

**There is no Info.plist in the framework created by Bazel**. And I don't see any way of including my own using the `ios_static_framework` rule. If it's required to run the app shouldn't it be there?

I also tried using `ios_framework` but it does not create the `modulemap` for me.

Am I doing something wrong? **How can I create a distributable binary using Bazel?**

**Comment by keith** [comment]
rules_apple does not provide a way to create a distributable xcframework https://github.com/bazelbuild/rules_apple/issues/355

If you don't need to support Apple Silicon you can build your `ios_static_framework` as a fat binary by only calling bazel once and passing `--ios_multi_cpus=x86_64,arm64` and using the output directly from Xcode.

> If it's required to run the app shouldn't it be there?

This isn't a requirement for static frameworks, you're likely pulling it in incorrectly. To integrate a static framework in Xcode you'll need to add the path to it to your `FRAMEWORK_SEARCH_PATHS` and then add it to a "Link binary with libraries" build phase, but make sure you do not embed it in the consuming applications / extensions.

**Comment by jagobagascon** [comment]
I want to distribute it using Swift Package Manager, and AFAIK it only supports xcframework binaries, so I thought packing the created static frameworks with `xcodebuild -create-xcframework` would work.

I can make it work if I import it manually (just how you said), that could work but it's far from ideal. Our goal is to provide the binary through SwiftPM. Would manually adding an `Info.plist` do the trick?

**Comment by keith** [comment]
You'd have to try it. Having a plist for a static framework doesn't really make sense since the bundle isn't shipped with the target like dynamic frameworks are. So maybe that's a requirement from xcodebuild because it expects a dynamic framework?

You can follow https://github.com/bazelbuild/rules_apple/issues/355 for any progress on supporting creation of xcframeworks via these rules.

**Comment by jagobagascon** [comment]
I managed to make it work so I'm just adding it here in case it's useful for someone:

*Disclaimer: I have no idea what I'm doing, this feels hacky, and it's probably wrong so I'm probably not gonna use it and neither should you.... but it works so far.*

As you said adding an `Info.plist` made no sense, but when adding dependencies using SPM there's no way of changing the 'Do not embed' value. It looks like xcode is always expecting a dynamic binary (maybe? no idea) so to fix it I just removed the native module from "Link Binary With Libraries" and added it to "Dependencies" in my Build Phases.

I also had to remove any internal dependencies from my Swift modules to the Native module:
```swift
products: [
    .library(name: "SwiftCode", targets: ["SwiftCode"]),
    .library(name: "NativeCode", targets: ["NativeCode"])
],
.targets: [
    .target(name: "SwiftCode"), // This uses NativeCode, but if I add a dependency I get the error mentioned above
    .binaryTarget(
        name: "NativeCode",
        path: "NativeCode.xcframework"
    )
]
```
